contracts/WildcardSteward_v2.sol

```diff
@@ -257,77 +265,92 @@ contract WildcardSteward_v2 is Initializable {
         return foreclosureTimePatron(tokenPatron);
     }
 
+    function _collectLoyalty(uint256 tokenId) internal {
+        address currentOwner = currentPatron[tokenId];
+        uint256 previousTokenCollection = timeLastCollected[tokenId];
+        uint256 patronageOwedByTokenPatron = patronageOwedPatron(currentOwner);
+        uint256 amountToMint;
+
+        if (patronageOwedByTokenPatron >= deposit[currentOwner]) {
+            uint256 newTimeLastCollected = timeLastCollectedPatron[currentOwner]
+                .add(
+                (
+                    (now.sub(timeLastCollectedPatron[currentOwner]))
+                        .mul(deposit[currentOwner])
+                        .div(patronageOwedByTokenPatron)
+                )
+            );
+            amountToMint = (newTimeLastCollected.sub(previousTokenCollection))
+                .mul(tokenGenerationRate[tokenId]);
+        } else {
+            amountToMint = now.sub(timeLastCollected[tokenId]).mul(
+                tokenGenerationRate[tokenId]
+            );
+
+        }
+        erc20Minter[tokenId].mint(currentOwner, amountToMint);
+        emit CollectLoyalty(tokenId, currentOwner, amountToMint);
```

Comment:
It is a pity that this code can't be done at the same time as collecting the patronage. But lets rather get things working first.

Reply:
Agree. Definitely something to put some thought into.